config/tectonic-node-controller/config.go

```diff
@@ -0,0 +1,40 @@
+package tnc
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Kind is the TypeMeta.Kind for the TNC config.
+	Kind = "TectonicNodeControllerConfig"
+	// APIVersion is the TypeMeta.APIVersion for the TNC config.
+	APIVersion = "v1"
+)
+
+// TectonicNodeControllerConfig is the config structure TNC to generate NodeConfigs
+type TectonicNodeControllerConfig struct {
+	metav1.TypeMeta       `json:",inline"`
+	HTTPProxy             string `json:"HTTPProxy"`
+	HTTPSProxy            string `json:"HTTPSProxy"`
+	NoProxy               string `json:"NoProxy"`
+	KubeletImageURL       string `json:"KubeletImageURL"`
+	KubeletImageTag       string `json:"KubeletImageTag"`
```

Comment:
Kubelet image/tag should be known explicitly in the tnc manifests - there should be no templating (so I think we can remove this).